static/app/utils/timeSeries/useFetchEventsTimeSeries.tsx

```diff
@@ -0,0 +1,94 @@
+import {normalizeDateTimeParams} from 'sentry/components/organizations/pageFilters/parse';
+import {encodeSort} from 'sentry/utils/discover/eventView';
+import type {Sort} from 'sentry/utils/discover/fields';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import type {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import type {TimeSeries} from 'sentry/views/dashboards/widgets/common/types';
+import type {SamplingMode} from 'sentry/views/explore/hooks/useProgressiveQuery';
+import {DEFAULT_SAMPLING_MODE} from 'sentry/views/insights/common/queries/useDiscover';
+import {
+  getRetryDelay,
+  shouldRetryHandler,
+} from 'sentry/views/insights/common/utils/retryHandlers';
+
+import {getIntervalForTimeSeriesQuery} from './getIntervalForTimeSeriesQuery';
+
+interface UseFetchEventsTimeSeriesOptions<Field> {
+  yAxis: Field | Field[];
+  enabled?: boolean;
+  groupBy?: Field[];
+  query?: MutableSearch;
+  sampling?: SamplingMode;
+  sort?: Sort;
+  topEvents?: number;
+}
```

Comment:
Very similar interface to `useDiscoverSeries`, but with some small changes:

1. Order of arguments is different
2. Some options are not supported (e.g., aliasing)
3. String queries not supported yet (probably soon, we'll see when we need it)